packages/tools/src/tools/annotation/planarFreehandROITool/interpolation/algorithms/bspline.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  interpolateBasis as d3InterpolateBasis,
+  quantize as d3Quantize,
+} from 'd3-interpolate';
+import { zip as d3Zip } from 'd3-array';
+import { Types } from '@cornerstonejs/core';
+
+/**
+ * Returns a interpolated segment (b-spline algorithm) based on knotsIndexes.
+ * KnotsIndexes tells the first/last index of segment and also which index will be present on interpolation.
```

Comment:
Would like more explanation here, as later developers may well want to know the intent of the algorithm.  Something like:
Interpolates a set of points indexed by knotsIndexes.  That is, it DISCARDS all points except those in knotsIndexes.  Then, a new set of points is created by using a bspline on the remaining points, in order to re-create a new set of points.   Returns an array of the originalPoints length containing the interpolated data.